BUILDING-configure.md

```diff
@@ -1,29 +1,64 @@
 Building using `configure`
 ==========================
 
-The build requires `libpq`, the C client library for PostgreSQL.  This library
-must be installed before you can build libpqxx.  You'll need the headers as
-well as the library binary.
-
 The instructions will assume that you're working from a command-line shell.
 If you prefer to work from an IDE, you'll have to know how your IDE likes to
 do things, and you'll want to follow the shell instructions as a guide.
 
+Prerequisites
+-------------
+
+Before you can begin the build process for libpqxx, you must first install
+`libpq-dev` and the latest `libpq5` library. These are C client libraries
```

Comment:
Gotcha.

The reason I put that there is that I got lost at this point, and I note based on this stackoverflow.com question that other people are getting lost, too.

https://stackoverflow.com/questions/63785462/problems-instaling-libpq-dev-in-ubuntu-20-04

There's a lot of competing information about how to install libpqxx, fyi. I probably spent many hours just to get it install and compile. If I wasn't on a paid project that needs this, I would have given up.

I'm not sure what the solution is. At least for ubuntu, I would prefer to have a bash script that installs the packages automatically. But that's beyond the scope of what I can offer right now, my bash skills aren't great enough.

Reply:
You make a strong case for fixing the documentation.  What if we added "for instance" notes?  "For instance, on Ubuntu 22.04 you would need to install `libpq-dev` and `postgresql-dev`."  And if we have to, I'd prefer to make a separate note about "how to install a package on your system" and say "for instance, on Ubuntu you'd `apt-get install <package>`."

That way we'd get short, orthogonal chunks of text.  Not the most convenient for the reader, but I really don't want to get bogged down again with more text than I can maintain well.

I'm surprised that your system still had a libpq 9 by the way — I think on Ubuntu you'd have to go back all the way to 16.04 to find that.  The oldest supported upstream version is 11.

Reply:
Your new paragraph says it well, I think.

However I dug in a bit deeper and found... the part about the headers is obsolete!  I _used_ to have to install `postgresql-server-dev-all` (not `postgresql-dev` as I wrote before) for one particular header, from which libpqxx needed just one `#define` line.  But eventually I just copied that line into my own headers (since it wasn't something that postgres could safely change anyway).  So now `libpq-dev` (on Debian/Ubuntu) should be enough.

Reply:
The best guesses I have as to why your distro defaulted to libpq 9 are: (a) the package maintainer for your OS simply didn't get around to updating it.  (As one possible reason, package maintainers sometimes take on too many packages and later get overwhelmed with the work.)  Or (b) the OS may have a slow release cycle and a conservative approach to compatibility.  Or (c) perhaps for some reason the 9 does not correspond to a PostgreSQL version.

PostgreSQL 9 is no longer even maintained upstream.

Reply:
Okay, I'm getting rid of the part about needing headers and binaries.

I'm asking around in the chat community for Pop!_OS about whether or not this is inherent in their distro, and to see if they can fix it. But we shouldn't get too hung up on this. It is possible that this issue is a one-off inherent in Pop and few other users of libpqxx are experiencing this hang up.

Reply:
I got a response from a pop dev who says that pop is simply based on the ubuntu jammy release, and that the issue is unlikely to be coming from the pop distro.

If this is just a one off event, I'd say we shouldn't worry about it. But if it happens again, ping me and I may be able to investigate some more.

Reply:
Strange.  To my knowledge, jammy ships with postgresql-14.  (That's the source package that produces libpq.)

Here's a wild bit of speculation...  Could something, somewhere, be set up to filter packages to look for an _x.y_ version number in the postgresql package name?  The last postgres versions that had the minor version number in there were 9.x.

Reply:
That is an interesting thought...

Do you know any way I could test my own dependencies? Some terminal command that would indicate why libpq-dev@9 would install, rather than the apt cache policy of 14.x? Any idea on how to phrase such a question? I wonder if there's something on my machine specifically that is preventing 14.x?

Reply:
Can't think of one...  But you could have a rummage around their package archive, and see what versions are in there.  They may have a Web UI for that, or you can just load up the archive urls from `/etc/apt/sources.list*` in a browser and look around.

Reply:
I'm sorry, I haven't been able to find anything on this. I did check in `/etc/apt/sources.list.d` and just saw some simple files. 

```
(base) bluesanta@zuko:2022-03-Lib-Science$ cd /etc/apt/sources.list.d/
(base) bluesanta@zuko:sources.list.d$ ls
google-chrome.list  pop-os-apps.sources  pop-os-release.sources  steam-beta.list  steam-stable.list  system.sources
(base) bluesanta@zuko:sources.list.d$ cat system.sources 
X-Repolib-Name: Pop_OS System Sources
Enabled: yes
Types: deb deb-src
URIs: http://apt.pop-os.org/ubuntu
Suites: jammy jammy-security jammy-updates jammy-backports
Components: main restricted universe multiverse
X-Repolib-ID: system
X-Repolib-Default-Mirror: http://apt.pop-os.org/ubuntu

(base) bluesanta@zuko:sources.list.d$ cat pop-os-apps.sources 
X-Repolib-Name: Pop_OS Applications
Enabled: yes
Types: deb
URIs: http://apt.pop-os.org/proprietary
Suites: jammy
Components: main
(base) bluesanta@zuko:sources.list.d$ cat pop-os-release.sources 
X-Repolib-Name: Pop_OS Release Sources
Enabled: yes
Types: deb deb-src
URIs: http://apt.pop-os.org/release
Suites: jammy
Components: main
```

Reply:
Yes, the main thing is probably in `/etc/apt/sources.list` itself.  The `sources.list.d` was added later so that various packages could add their own snippets, without modifying package-installed files that would then later complicate package upgrades with that question — "your installed version of the file has modifications,should we keep your version or use the package maintainer's version?"